tools/syz-declextract/syz-declextract.cpp

```diff
@@ -64,6 +106,230 @@ bool beginsWith(const std::string_view &str, const std::string_view begin) {
 
 bool contains(const std::string_view &str, const std::string_view sub) { return str.find(sub) != std::string::npos; }
 
+bool isPadding(const std::string &fieldName) {
+  return contains(fieldName, "pad") || contains(fieldName, "unused") || contains(fieldName, "_reserved");
+}
+
+const std::string makeArray(const std::string &type, const size_t len = 0) {
+  if (len == 1) {
+    return type;
+  }
+  if (len) {
+    return "array[" + type + ", " + std::to_string(len) + "]";
+  }
+  return "array[" + type + "]";
+}
+
+const std::string makeArray(const std::string &type, const size_t min, const size_t max) {
+  if (min == max) {
+    return makeArray(type, min);
+  }
+  return "array[" + type + ", " + std::to_string(min) + ":" + std::to_string(max) + "]";
+}
+
+const std::string makePtr(const std::string &dir, const std::string &type, bool isOpt = false) {
+  if (type == "void") {
+    return makePtr(dir, makeArray("int8"));
+  }
+  std::string ptr = "ptr[" + dir + ", " + type;
+  if (isOpt) {
+    return ptr + ", opt]";
+  }
+  return ptr + "]";
+}
+
+const std::string makeConst(const std::string &type, const std::string &val = "0") {
+  if (type.empty()) {
+    return "const[" + val + "]";
+  }
+  if (type == "void") {
+    return "const[" + val + ", intptr]";
+  }
+  return "const[" + val + ", " + type + "]";
+}
+
+const std::string getSyzType(const std::string &ctype, const bool isSyscallParam = false) {
+  if (!isSyscallParam && (contains(ctype, "long long") || contains(ctype, "64"))) {
+    return contains(ctype, "be") ? "int64be" : "int64";
+  }
+  if (contains(ctype, "16") || contains(ctype, "short")) {
+    return contains(ctype, "be") ? "int16be" : "int16";
+  }
+  if (contains(ctype, "8")) {
+    return contains(ctype, "be") ? "int8be" : "int8";
+  }
+  if (contains(ctype, "32") || contains(ctype, "int")) {
+    return contains(ctype, "be") ? "int32be" : "int32";
+  }
+  if (contains(ctype, "void")) {
+    return "void";
+  }
+  if (contains(ctype, "char")) {
```

Comment:
This would be reasonable to move to `contains(ctype, "8")` case.
You have `if (contains(ctype, "16") || contains(ctype, "short"))`, so it's reasonable to have `if (contains(ctype, "8") || contains(ctype, "char"))`.

Reply:
The reasoning was that some types are typedef'd as char16 or char32, but currently getSyzType only gets called from a builtin type, so removing it should not cause any differences.